### PR TITLE
Set Up Associations

### PR DESCRIPTION
**Pull request by demix007**
## In this milestone, I implemented the following:

- [ ] Creation of a class Classroom with the following:

    💠 @label instance variable, that is initialized in the constructor.
    💠 Setter and getter for @label.

- [ ] Creation of the has-many/belongs-to relationship between Classroom and Student. The following were implemented:

    💠 Creation of the has-many side (a classroom has many students).
    💠Creation of the belongs-to side (a student belongs to a classroom).
    💠Made sure that when adding a student to a classroom it also sets the classroom for the student.
    💠Made sure that when setting the classroom for a student it also adds it to the classrooms' students.

- [ ] Creation of a class Book with the following:

    💠@title and @author instance variables which were initialized in the constructor.
    💠Setters and getters for instance variables.

- [ ] Creation of a class Rental with the following:

    💠@date instance variable, that was initialized in the constructor.
    💠Setter and getter for @date.

- [ ] Creation of the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following were implemented:

    💠Creation of the has-many side of Book and Rental (a book has many rentals).
    💠Creation of the belongs-to side of Rental and Book (a rental belongs to a book).
    💠Creation of the has-many side of Person and Rental (a person has many rentals).
    💠Creation of the belongs-to side of Rental and Person (a rental belongs to a person).
    💠Modification of the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :label
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @student.push(student)
```
Comment (DammyShittu) [inline classroom.rb]: - [x] This should be `@students.push(student)` based on what you defined in the `initialize` method. Kindly fix that.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date)
+    @date = date
+    @book = book
+    book.rental_details << self
+    @person = person
+    person.rental_details << self
+  end
```
Comment (DammyShittu) [inline rental.rb]: - [x] You are using `book` and `person` objects without passing them to the constructor. Kindly include `book` and `person` in the parameters.

  ```
  def initialize(date, book, person)
    ...
  end
  ```

File: person.rb [hunk 3]
```diff
@@ -22,4 +23,9 @@ def can_use_service?
   def correct_name
     @name
   end
+
+  def add_rental_details(date, book)
+    Rental.new(date, self, book)
+    rental_details.push(self)
+  end
```
Comment (DammyShittu) [inline person.rb]: - [x] The code on line 28 needs to match the Rental constructor parameter order. You should have this depending on the order of the parameters.

  ```
  def add_rental_details(date, book)
    Rental.new(date, book, self)
  end
  ```

- [x] Kindly remove the code on line 29. Initializing a rental on line 28 already adds the rental to the `rental_details` array in the Rental class, so there is not need to add a duplicate.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @demix007,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (demix007) [reply classroom.rb]: Thank you for bringing my attention to this, the change has been implemented.

File: rental.rb (hunk 2 above)
Comment (demix007) [reply rental.rb]: I have also implemented the change which was required here.

File: person.rb (hunk 3 above)
Comment (demix007) [reply person.rb]: Thanks for this wonderful insight, I have made changes to both line 28 and 29 just as required.

**Comment by demix007** [comment]
## Hello @DammyShittu 👋🏼.

Thank you for taking out time to review and recommend changes to this PR.  The changes have been implemented just as required. 💯

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @demix007 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by demix007** [comment]
# Hello @bobb-Rob 👋🏼 ,

Thank you for taking out time to review and approve this PR. You are awesome!

**Pull request merged**